### ISSUE TITLE
Dictionary key sorting in yaml dump cannot be disabled on rhel-8

### ISSUE THREAD

**Issue by psss** [psss/tmt#207]
The latest copr build [failed](https://download.copr.fedorainfracloud.org/results/psss/tmt/epel-8-x86_64/01338870-tmt/builder-live.log.gz) on rhel-8 becase `python3-pyyaml` is old and does not support turning of dictionary key sorting when dumping data into yaml. We do we do? Work around that for rhel-8 to store dictionaries sorted? This will produce ugly `main.fmf` when doing for example `tmt test import`. Also `tests.yaml` will have tests sorted by name and there might be more. Or push for a newer version of the package? Or both? @thrix, @lukaszachy, @pvalena, what do you say?

**Comment by psss** [comment]
Simple reproducer:

```python
import yaml
import io

data = dict(one=1, two=2, three=3)
print(data.keys())

output = io.StringIO()
yaml.safe_dump(
    data, output, sort_keys=False,
    encoding='utf-8', allow_unicode=True,
    width=1000000, indent=4, default_flow_style=False)

print(output.getvalue())
```

python3-pyyaml-3.12-12.el8.x86_64

```
dict_keys(['one', 'two', 'three'])
Traceback (most recent call last):
  File "ha.py", line 11, in <module>
    width=1000000, indent=4, default_flow_style=False)
  File "/usr/lib64/python3.6/site-packages/yaml/__init__.py", line 216, in safe_dump
    return dump_all([data], stream, Dumper=SafeDumper, **kwds)
TypeError: dump_all() got an unexpected keyword argument 'sort_keys'
```

python3-pyyaml-5.3.1-1.fc31.x86_64

```
dict_keys(['one', 'two', 'three'])
one: 1
two: 2
three: 3
```

**Comment by lukaszachy** [comment]
In [rhel-8.2 ](https://www.redhat.com/en/blog/red-hat-enterprise-linux-82-beta-now-available)we will have python-3.8 which contains newer pyyaml. 
IMHO we should be using that (for next 3 years or how long is module supported)

**Comment by psss** [comment]
Ok, so here's a temporary [workaround](https://stackoverflow.com/questions/31605131/) before `rhel-8.2` is released:

```python
representer = lambda self, data: self.represent_mapping(
    'tag:yaml.org,2002:map', data.items())
yaml.add_representer(dict, representer)
```

Once the `python38` is available we cat get rid of it.
